pyccl/pk2d.py

```diff
@@ -1,13 +1,27 @@
 import warnings
-
+import functools
 import numpy as np
 
 from . import ccllib as lib
-
 from .errors import CCLWarning
 from .pyutils import check, _get_spline2d_arrays
 
 
+class _Pk2D_descriptor:
+    """Descriptor to enable usage of `Pk2D` class methods as instance methods.
+    """
+    def __init__(self, func):
+        self.func = func
+
+    def __get__(self, instance, base):
+        this = instance if instance else base
+
+        @functools.wraps(self.func)
+        def new_func(*args, **kwargs):
+            return self.func(this, *args, **kwargs)
+        return new_func
+
+
 class Pk2D(object):
     """A power spectrum class holding the information needed to reconstruct an
     arbitrary function of wavenumber and scale factor.
```

Comment:
I think we don't actually have this anywhere: could you add a couple of lines to the docstring informing people that they can add/multiply etc. Pk2Ds?

Reply:
done